pkgs/development/python-modules/nltk/default.nix

```diff
@@ -1,17 +1,21 @@
-{ fetchurl, buildPythonPackage, lib, six, pythonAtLeast, pythonOlder }:
+{ fetchPypi, buildPythonPackage, lib, six, singledispatch, isPy3k }:
 
 buildPythonPackage rec {
-  version = "3.2.5";
+  version = "3.4";
   pname = "nltk";
 
-  src = fetchurl {
-    url = "mirror://pypi/n/nltk/nltk-${version}.tar.gz";
-    sha256 = "2661f9971d983db314bbebd51ba770811a362c6597fd0f303bb1d3beadcb4834";
+  src = fetchPypi {
+    inherit pname version;
+    extension = "zip";
+    sha256 = "286f6797204ffdb52525a1d21ec0a221ec68b8e3fa4f2d25f412ac8e63c70e8d";
   };
 
-  propagatedBuildInputs = [ six ];
+  propagatedBuildInputs = [ six ] ++ lib.optional (!isPy3k) singledispatch;
 
-  disabled = pythonOlder "2.7" || pythonOlder "3.4" && (pythonAtLeast "3.0");
+  # TODO: remove patch during update to new version
+  patchPhase = ''
+  sed -ie 's/singledispatch/singledispatch; python_version < \\"3.4\\"/g' setup.py
+  '';
```

Comment:
Please use `fetchpatch` with https://github.com/nltk/nltk/commit/3966111cbf2f35fb86082b2f12acd90d75e9b8bb.patch.

Reply:
Do we really need this? What a point?

Reply:
It is way more robust than a sed. If the patch does not apply, the build fails. If your sed breaks something, we end up with a strange error output to debug - or worse a half-working package.

Reply:
Got it. I'll fix soon. Thank you.

Reply:
@dotlambda @matthiasbeyer https://github.com/NixOS/nixpkgs/pull/59204/commits/2e5aceadb86c32f391708930e6ec17d5076825ea Please check